eZ/Bundle/EzPublishCoreBundle/SiteAccess/SiteAccessMatcherRegistry.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Bundle\EzPublishCoreBundle\SiteAccess;
+
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+
+final class SiteAccessMatcherRegistry
+{
+    /** @var \eZ\Bundle\EzPublishCoreBundle\SiteAccess\Matcher[] */
+    private $matchers;
+
+    /**
+     * @param \eZ\Bundle\EzPublishCoreBundle\SiteAccess\Matcher[] $matchers
+     */
+    public function __construct(array $matchers = [])
+    {
+        $this->matchers = $matchers;
+    }
+
+    public function setMatcher(string $identifier, Matcher $matcher): void
```

Comment:
`addMatcher` would be a better name imo

Reply:
Actually, based on class purpose and name, maybe rather `registerMatcher`?

Reply:
I take it back, `set` is proper name